refactor(core): Add more unit tests for `Workflow.ts` (no-changelog)

netroy opened the pull request:
## Summary
Extracted out of #9858

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included

netroy committed:
refactor: group the tests for renameNodeInParameterValue. no actual code change

netroy committed:
add unit tests for checkIfWorkflowCanBeActivated, and fix soe linting error

netroy committed:
add unit tests for checkReadyForExecution

despairblue commented on packages/workflow/test/Workflow.test.ts:
```diff
@@ -68,6 +69,84 @@ describe('Workflow', () => {
+				nodeTypes,
+			});
+
+			const issues = workflow.checkReadyForExecution({ startNode: disabledNode.name });
```
You're given a `startNode` in multiple tests here, but no `destinationNode`.
All these tests would also work if we didn't pass the start node. So I assume it's to be able to test the branch that calls `getChildNodes`.
Would it make sense to change one of these tests to pass `destinationNode` so that we also cover `getParentNodes`?

netroy replied:
I think we should add a separate group of tests for `destinationNode`. 
But, I can do that in the next PR adding tests for `Workflow`.

despairblue committed:
improve checkIfWorkflowCanBeActivated tests for ignored nodes and disabled nodes

1. ignore a node that would otherwise make the function return true
2. disable a node that would make the function return true if it wasn't
disabled

despairblue reviewed (commented):
I have one question, the other 2 things I already resolved with a commit.

janober commented:
Got released with [`n8n@1.48.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.48.0)
